### PR TITLE
modules/tectonic/resources/tectonic.sh: update wait for pods logic

### PR DESCRIPTION
**Pull request by rithujohn191**
Account for the case when `kubectl` command returns no resources

### CODE REVIEW

**Comment by Quentin-M** [comment]
ok to test

**Comment by rithujohn191** [comment]
ok to test

**Comment by s-urbaniak** [comment]
ok to test

File: modules/tectonic/resources/tectonic.sh [hunk 1]
```diff
@@ -56,15 +56,36 @@ function wait_for_tpr() {
...
+  
+    if [ "$status" -ne "0" ]; then
+      echo "kubectl command failed, retrying in 5 seconds"
+      sleep 5
+      continue
+    fi
+  
+    # make sure kubectl does not return "no resources found"
+    if [ "$(echo "$out" | tail -n +2 | grep -c '^')" -eq 0 ]; then
+      echo "no nodes were found, retrying in 5 seconds"
```
Comment (s-urbaniak) [inline modules/tectonic/resources/tectonic.sh]: nit: `no resources were found, retrying in 5 seconds`

File: modules/tectonic/resources/tectonic.sh [hunk 2]
```diff
@@ -56,15 +56,36 @@ function wait_for_tpr() {
...
+      sleep 5
+      continue
+    fi
+  
+    stat=$( echo "$out"| tail -n +2 | grep -v '^Running')
```
Comment (s-urbaniak) [inline modules/tectonic/resources/tectonic.sh]: - nit (remove the extraneous whitespace) in `$( echo...`
- additionally we have to wrap this call with `set +e`/`set -e`. `grep` will return with exit code 1, causing the whole script to fail. I think this is also the cause for the CI failure.

File: modules/tectonic/resources/tectonic.sh (hunk 1 above)
Comment (rithujohn191) [inline modules/tectonic/resources/tectonic.sh]: Done

File: modules/tectonic/resources/tectonic.sh (hunk 2 above)
Comment (rithujohn191) [inline modules/tectonic/resources/tectonic.sh]: Done. Thought it would be more clean to `set +e/set -e` for the whole function.

**Comment by rithujohn191** [comment]
ok to test

**Comment by s-urbaniak** [comment]
ok to test

File: modules/tectonic/resources/tectonic.sh (hunk 2 above)
Comment (s-urbaniak) [inline modules/tectonic/resources/tectonic.sh]: @rithujohn191 agreed 👍

**Pull request merged**